connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -372,6 +374,22 @@ class SparkConnectPlanner(session: SparkSession) {
     }
   }
 
+  private def transformCollectMetrics(rel: proto.CollectMetrics): LogicalPlan = {
+    val metrics = rel.getMetricsList.asScala.toSeq.map { expr =>
+      Column(transformExpression(expr))
+    }
+
+    if (rel.getIsObservation) {
+      val observation = Observation(rel.getName)
+      Dataset
+        .ofRows(session, transformRelation(rel.getInput))
+        .observe(observation, metrics.head, metrics.tail: _*)
+        .logicalPlan
+    } else {
+      CollectMetrics(rel.getName, metrics.map(_.named), transformRelation(rel.getInput))
```

Comment:
Does this actually make sense? Looking the way this class is used it seems that only `.observe` actually creates an instance of `CollectMetrics`. I'm not sure we're actually exposing this operator as such.

@hvanhovell what's your perspective?

Reply:
Many other api exposing operator directly.

Reply:
Can you please give me an example of where we're exposing the catalyst operator directly in our API (in particular in the Dataset API)?

Reply:
For example, `Unpivot`, `UnresolvedHint` and so on.

Reply:
As outlined above, only this branch is needed, their behavior is identical

Reply:
The above branch will cache the logical plan but the below branch do not.

Reply:
I am sorry, but where is the logicalplan cached in all of this?

Reply:
@hvanhovell I'm sorry. I also supports `df.randomSplits`. The reply just now is confused. `df.observe` without cache behavior. We can only support string now.